src/main/java/com/dabsquared/gitlabjenkins/cause/CauseData.java

```diff
@@ -115,7 +115,7 @@ public final class CauseData {
         this.actionType = Objects.requireNonNull(actionType, "actionType must not be null.");
         this.sourceProjectId = Objects.requireNonNull(sourceProjectId, "sourceProjectId must not be null.");
         this.targetProjectId = Objects.requireNonNull(targetProjectId, "targetProjectId must not be null.");
-        this.projectId = Objects.requireNonNull(projectId, "projectId must not be null.");
+        this.projectId = projectId == null ? 0 : projectId;
```

Comment:
```suggestion
        this.projectId = Objects.requireNonNull(projectId, "projectId must not be null.");
```

We should not change the original logic for handling the `projectId` in "src/main/java/com/dabsquared/gitlabjenkins/cause/CauseData.java"